lms/djangoapps/courseware/tests/test_entrance_exam.py

```diff
@@ -206,6 +211,100 @@ def setUp(self):
             ]
         )
 
+    @mock.patch('xmodule.x_module.XModuleMixin.has_dynamic_children', mock.Mock(return_value='True'))
+    def test_view_redirect_if_entrance_exam_required(self):
```

Comment:
@mattdrayer Should this PR prevent you from directly hitting a portion of the course that is "beyond" the entrance exam? On the sandbox, I can go directly to http://pr6997.m.sandbox.edx.org/courses/wew/wefe/wewe/courseware/cad30653cd2b44bcafde2d34a959897b/618fd05fe56347d4808a74d3b7ae1936/1 (logged in as verified@example.com), even though I have not finished the entrance exam.